### PR TITLE
Update APIs based on consistency review

### PR DESCRIPTION
**Pull request by srnagar**
Made some changes to the API to be consistent across languages:
- Consolidated separate setter methods for additional workspaces into a single collection
- Enabled getting log response as a model at `LogsQueryResult` and `LogsTable` level
- Updated samples and tests

### CODE REVIEW

File: sdk/monitor/azure-monitor-query/README.md [hunk 1]
```diff
@@ -56,166 +56,57 @@ LogsQueryAsyncClient logsQueryAsyncClient = new LogsQueryClientBuilder()
...
-
-        public Object getOperationLinks() {
-            return operationLinks;
-        }
-
-        public void setOperationLinks(Object operationLinks) {
-            this.operationLinks = operationLinks;
-        }
-    }
+List<CustomModel> results = queryResults.getResultAsObject(CustomModel.class);
```
Comment (pallavit) [inline sdk/monitor/azure-monitor-query/README.md:100]: This is a really good optimization!

File: sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/LogsQueryAsyncClient.java [hunk 2]
```diff
@@ -148,28 +140,11 @@
                 .map(this::convertToLogQueryBatchResult);
     }
 
-    private String buildPreferHeaderString(LogsQueryOptions query) {
-        StringBuilder sb = new StringBuilder();
-        if (query.isIncludeRendering()) {
-            sb.append("include-render=true");
+    private Context updateContext(Duration serverTimeout, Context context) {
```
Comment (pallavit) [inline sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/LogsQueryAsyncClient.java:143]: >updateContext [](http://example.com/codeflow?start=20&length=13)

Isn't context a reference type? Do we need to return it?

File: sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/models/LogsQueryOptions.java [hunk 3]
```diff
@@ -92,98 +73,38 @@ public LogsQueryOptions setIncludeStatistics(boolean includeStatistics) {
...
-     */
-    public LogsQueryOptions setWorkspaceNames(List<String> workspaceNames) {
-        this.workspaceNames = workspaceNames;
-        return this;
-    }
-
-    /**
-     * Returns the list of workspace ids on which this query is executed.
-     * @return The list of workspace ids on which this query is executed.
+     * Returns the list of additional workspaces on which this query is executed. The list can contain any of the
```
Comment (pallavit) [inline sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/models/LogsQueryOptions.java:76]: Curious - why we made this change? So, earlier workspace was always part of options while now client is directly associated with at least a workspace?

File: sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/LogsQueryAsyncClient.java (hunk 2 above)
Comment (srnagar) [reply sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/LogsQueryAsyncClient.java:143]: No, context is like a linked list. When new data is added, the head location changes and we need to use the new head reference. So, we have to return the context with updated reference. See [example](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core-tracing-opentelemetry/src/main/java/com/azure/core/tracing/opentelemetry/OpenTelemetryTracer.java#L146)

File: sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/models/LogsQueryOptions.java (hunk 3 above)
Comment (srnagar) [reply sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/models/LogsQueryOptions.java:76]: This change was made to consolidate 4 different types of lists into a single list as they are all strings and it isn't necessary to know whether they are ids, names or resourceIds.

Comment (pallavit) [reply sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/models/LogsQueryOptions.java:76]: So, one of the core scenarios for the SDK is to run the same query on different workspaces. Why would one want to do it?

**Review by pallavit — APPROVED** [review]

File: sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/models/LogsQueryOptions.java (hunk 3 above)
Comment (srnagar) [reply sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/models/LogsQueryOptions.java:76]: It is not a core scenario but the service supports querying multiple workspaces using the same query. One example I can think of is if someone wants to look at all the HTTP requests in all the workspaces they own, they could use this feature. Each workspace is like a logical grouping of related data. So, customers can choose to have 1 workspace per geographical region - NA, Asia-Pacific, Europe etc and to get global results, they run queries across all these regions (workspaces).

**Pull request merged**